hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,53 @@
+pulp_db_param = param('katello', 'pulp_db_name') || param('foreman_proxy_content', 'pulp_db_name')
+fail_and_exit('Unknown pulp database name') unless pulp_db_param
+MONGO_DB_NAME = pulp_db_param.value
+
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def migration
+  export_dir = '/var/tmp/engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+  mongo_backup = '/var/tmp/mongo_backup'
+  mongo_conf = '/etc/opt/rh/rh-mongodb34/mongod.conf'
+
+  # Create export directory and dump MongoDB
+  logger.info 'Stopping Pulp services except MongoDB'
+  Kafo::Helpers.execute('systemctl stop rh-mongodb34-mongod')
+  FileUtils.mkdir_p(export_dir)
+  File.chmod(0700, export_dir)
+  logger.info "Starting mongodump to #{export_dir}"
+  Kafo::Helpers.execute("mongodump --host localhost --out #{export_dir}")
+
+  # Move datafiles out of MongoDB directory so it will start
+  logger.info 'Export done, stopping MongoDB to remove old datafiles'
+  Kafo::Helpers.execute('systemctl stop rh-mongodb34-mongod')
+  logger.info "Moving contents from #{mongo_dir} to #{mongo_backup}"
+  FileUtils.mkdir_p(mongo_backup)
+  File.chmod(0700, mongo_backup)
+  Kafo::Helpers.execute("mv #{mongo_dir}/* #{mongo_backup}")
+
+  # Import the dump, fail and notify user of backup if restore does not work.
+  logger.info 'Changing config to WiredTiger and starting restore.'
+  Kafo::Helpers.execute("sed -i -e 's/mmapv1/wiredTiger/g' #{mongo_conf}")
+  Kafo::Helpers.execute("mongorestore --host localhost --db #{MONGO_DB_NAME} --quiet --dir #{export_dir}")
```

Comment:
Does this work if the database is stopped?

Reply:
Good catch, just tried here and yeah it fails:

~~~
[root@centos7-katello-3-7 ~]# mongorestore dump/pulp_database/
2018-06-25T18:44:11.888+0000	Failed: error connecting to db server: no reachable servers
~~~